### PR TITLE
Level system

### LINKED ISSUE #43: Level up/exp mechanic with persistence between games
Allow the player to level up, display the player's level and EXP requirements for next level in the UI, and persist the player's level between games with an option to reset on the main menu.

This will form the basis for skill tree upgrades and save slots.

### PR DESCRIPTION
**Pull request by shraymittal**
- Adds a persistent leveling and exp mechanic to the game with game save.
- Changes movement from move_and_collide to move_and_slide
- Makes player movement slightly more responsive with faster acceleration and no accel requirements at max_velocity adjacent directions

XP is equal to total score across all games.
EXP req per level is currently 500 * level^2 - 500 * level

These can be tuned as necessary going forwards.

Closes #43 and #45

### CODE REVIEW

File: BismarkUnchained/scenes/Player/Player.gd [hunk 1]
```diff
@@ -35,18 +37,22 @@ func move_translate(delta):
...
+		new_velocity = velocity
+	
 	# Checking that new velocity is not greater magnitude than target velocity, and prevents stuttering at extremely low speeds
 	if new_velocity.length() > velocity.length():
 		new_velocity = velocity
 	
 	# Executing the movement, setting current velocity for next pass, setting global time_delta
 	velocity = new_velocity
-	move_and_collide(velocity * delta)
+	move_and_slide(velocity)
```
Comment (Noviv) [inline BismarkUnchained/scenes/Player/Player.gd:55]: why not `velocity * delta`?

File: BismarkUnchained/scenes/Player/Player.gd [hunk 2]
```diff
@@ -74,6 +80,32 @@ func shoot():
...
+	return 500 * pow(next_level, 2) - 500 * next_level
+
+func save():
+	var save_dict = {
+		"experience" : experience,
+	}
+	var save_file = File.new()
+	save_file.open("res://bismarkunchained.sav", File.WRITE)
+	save_file.store_line(to_json(save_dict))
+	save_file.close()
```
Comment (Noviv) [inline BismarkUnchained/scenes/Player/Player.gd:100]: does this append or overwrite?

**Review by Noviv — APPROVED** [review]
left some comments you might want to look at

also have we decided between experience and level points? experience points would be continuous, and level points would be granted upon leveling up? I'm thinking that if we only level up between games (not between waves) experience points might be the best - not sure why we even need levels then

File: BismarkUnchained/scenes/Player/Player.gd (hunk 1 above)
Comment (shraymittal) [reply BismarkUnchained/scenes/Player/Player.gd:55]: Move_and_slide automatically takes into account delta

File: BismarkUnchained/scenes/Player/Player.gd (hunk 2 above)
Comment (shraymittal) [reply BismarkUnchained/scenes/Player/Player.gd:100]: Overwrite

**Pull request merged**